src/lib.rs

```diff
@@ -1134,12 +1130,12 @@ impl<TRepoTypes: RepoTypes> Future for IpfsFuture<TRepoTypes> {
                         };
                         let _ = ret.send(addrs);
                     }
-                    IpfsEvent::GetProviders(key, ret) => {
-                        let future = self.swarm.get_providers(key);
+                    IpfsEvent::GetProviders(cid, ret) => {
+                        let future = self.swarm.get_providers(cid);
                         let _ = ret.send(future);
                     }
-                    IpfsEvent::Provide(key, ret) => {
-                        let _ = ret.send(self.swarm.start_providing(key));
+                    IpfsEvent::Provide(cid, ret) => {
+                        let _ = ret.send(self.swarm.start_providing(cid));
```

Comment:
I think we should have the key before providing it (as in: get_block) or error if we dont have it?

At least a comment would be needed here.

Reply:
Oops sorry it was handled above already.